### PR TITLE
feat(cli): add traefik to cli

### PR DESCRIPTION
**Pull request by c3ppc3pp**
- added Traefik to CLI
- added ExtraHosts to HostsConfig in builder, to remove the unreliable and platform dependent network IP "hack" from the CLI (this also adds the ability to address the host machine)
- removed network gateway IP queries
- added internal host domain use
- fixed unhandled error
- fixed url issues where it would show the containers domain instead of "localhost"
- fixed routing issues where you have to access multiple ports, now everything is routed through traefik where needed by default
- added template for traefik.yaml file

### CODE REVIEW

File: cli/container_defaults.go [hunk 1]
```diff
@@ -123,11 +132,69 @@ func GetCruxUI(settings *Settings) *containerbuilder.DockerContainerBuilder {
...
+		commands = append(commands, "--providers.file.filename=/etc/traefik.dev.yml")
+	}
+
+	traefik := containerbuilder.NewDockerBuilder(context.Background()).
+		WithImage("docker.io/library/traefik:v2.8").
+		WithName(settings.Containers.Traefik.Name).
+		WithRestartPolicy(containerbuilder.AlwaysRestartPolicy).
+		WithoutConflict().
+		WithForcePullImage().
+		// WithEnv([]string{
```
Comment (robot9706) [inline cli/container_defaults.go:175]: Comments 🙀

**Review by robot9706 — CHANGES_REQUESTED** [review]
Comment + testing in progress 🏗

File: cli/container_defaults.go (hunk 1 above)
Comment (c3ppc3pp) [reply cli/container_defaults.go]: done, thanks!

File: cli/runner.go [hunk 2]
```diff
@@ -140,6 +172,61 @@ func StopContainers(containers *DyrectorioStack) {
...
+	if containerID := containers.Containers.Traefik.Name; GetContainerID(containerID) != "" {
+		CleanupContainer(containerID)
+	}
+}
+
+// Copy to Traefik Container
+func TraefikConfiguration(name, internalHostDomain string, cruxuiport uint) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		log.Fatalf("error: %v", err)
```
[the remark this answers is not in the document]
Comment (c3ppc3pp) [reply cli/runner.go]: Added more verbosity, in this case the underlying docker SDK should provide proper error message.

File: cli/runner.go [hunk 3]
```diff
@@ -140,6 +172,61 @@ func StopContainers(containers *DyrectorioStack) {
...
+		cli,
+		name,
+		"traefik.dev.yml",
+		data,
+		int64(len([]rune(result.String()))),
+		strings.NewReader(result.String()),
+	)
+
+	if err != nil {
+		log.Fatalf("error: %v", err)
```
[the remark this answers is not in the document]
Comment (c3ppc3pp) [reply cli/runner.go]: Added more verbosity, in this case the underlying docker SDK should provide proper error message as the dagentutils package doesn't provide its own error messages.

**Review by polaroi8d — APPROVED** [review]
LGTM (informal), tested in local and works like a charm 🦄

File: cli/runner.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (c3ppc3pp) [reply cli/runner.go]: We talked about this earlier, I redid the formatting, should report errors like you described.

File: cli/main.go [hunk 4]
```diff
@@ -9,6 +9,8 @@ import (
 	cli "github.com/urfave/cli/v2"
 )
 
+const errmsg = "%s: %v"
```
Comment (nandor-magyar) [inline cli/main.go:12]: This doesn't make the code more readable, rather the opposite.

**Comment by nandor-magyar** [comment]
Please rebase.

**Review by robot9706 — APPROVED** [review]
🚀

**Pull request merged**